src/scripts/main.js

```diff
@@ -1,3 +1,77 @@
 'use strict';
 
-// write code here
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+const tBody = document.querySelector('tbody');
+
+const fnAppendRow = function() {
+  if (tBody.children.length < 10) {
+    const row = document.querySelector('tr');
+    const rowCopy = row.cloneNode(true);
+
+    tBody.append(rowCopy);
+
+    if (tBody.children.length > 2) {
+      removeRow.removeAttribute('disabled');
+    }
+  } else {
+    appendRow.setAttribute('disabled', true);
+  }
+};
+
+const fnRemoveRow = function() {
+  if (tBody.children.length > 2) {
+    tBody.lastElementChild.remove();
+
+    if (tBody.children.length < 10) {
+      appendRow.removeAttribute('disabled');
+    }
+  } else {
+    removeRow.setAttribute('disabled', true);
+  }
+};
+
+const fnAppendColumn = function() {
+  const rows = document.querySelectorAll('tr');
+
+  for (const row of Array.from(rows)) {
+    if (row.children.length < 10) {
+      const td = document.createElement('td');
+
+      row.append(td);
+
+      if (row.children.length > 2) {
```

Comment:
Replace `row.children.length >= 2` for disable `-` button when we have 2 columns. Check other similar cases.